src/main/java/Storage.java

```diff
@@ -0,0 +1,201 @@
+                    }
+                } else {
+                    if (i == TaskList.size()) {
+                        s = s + scanFile.nextLine();
+                    } else {
+                        s = s + scanFile.nextLine() + System.lineSeparator();
+                    }
+                }
+            }
+            FileWriter fw = new FileWriter("data/duke.txt");
+            fw.write(s);
+            fw.close();
+        } catch (IndexOutOfBoundsException e) {
+            System.out.println("Invalid task number.");
+        } catch (FileNotFoundException e) {
+            System.out.println("File does not exist");
+        } catch (IOException e) {
+            System.out.println(e.getMessage());
+        }
+    }
+
+    /**
+     * Loads the tasks in the file into an ArrayList of tasks
+     *
+     * @return An ArrayList containing tasks from the file.
+     */
+    public ArrayList<Task> load() {
+        ArrayList<Task> tasks = new ArrayList<>();
+        try {
+            Scanner sc = new Scanner(file);
+            while (sc.hasNext()) {
+                String type = sc.next();
+                switch (type) {
+                case "T":
+                    String todo = sc.nextLine().trim();
+                    String[] todoArray = todo.split("\\|");
+                    Task todoTask = new Todo(todoArray[2].trim());
+                    if (todoArray[1].trim().equals("1")) {
+                        todoTask.markAsDone();
+                    }
+                    tasks.add(todoTask);
+                    break;
+                case "D":
+                    String deadline = sc.nextLine().trim();
+                    String[] deadlineArray = deadline.split("\\|");
+                    Task deadlineTask = new Deadline(deadlineArray[2].trim(), Parser.convertDateAndTime(deadlineArray[3].trim()));
+                    if (deadlineArray[1].trim().equals("1")) {
+                        deadlineTask.markAsDone();
+                    }
+                    tasks.add(deadlineTask);
+                    break;
+                case "E":
+                    String event = sc.nextLine().trim();
+                    String[] eventArray = event.split("\\|");
+                    Task eventTask = new Event(eventArray[2].trim(), Parser.convertDateAndTime(eventArray[3].trim()));
+                    if (eventArray[1].trim().equals("1")) {
+                        eventTask.markAsDone();
+                    }
+                    tasks.add(eventTask);
+                    break;
+                }
+            }
+        } catch (FileNotFoundException e) {
+            System.out.println("File does not exist");
+        }
+        return tasks;
+    }
+
+    /**
+     * Add the task of type Todo to the file.
+     *
+     * @param taskname A string representing the task description.
+     */
+    public void addTodo(String taskname) {
```

Comment:
_Suggestion \:_
Generalise the logic when adding a new task to avoid adding methods every time a new type of Task is added.
For example (pseudo code) \:
    `fileWriter.write(String.format("%s | %s | %s", task.getType().toString(), task.isDone(), task.getDescription());`